src/fr/anisama/src/eu/kanade/tachiyomi/animeextension/fr/anisama/AniSama.kt

```diff
@@ -0,0 +1,345 @@
+        title = element.select(".dynamic-name").text()
+        thumbnail_url = element.select(".film-poster-img").attr("data-src")
+        setUrlWithoutDomain(element.select(".film-poster-ahref").attr("href"))
+    }
+
+    override fun popularAnimeNextPageSelector() = ".ap__-btn-next a:not(.disabled)"
+
+    override fun popularAnimeParse(response: Response): AnimesPage {
+        aniSamaFilters.fetchFilters()
+        return super.popularAnimeParse(response)
+    }
+
+    // =============================== Latest ===============================
+    override fun latestUpdatesRequest(page: Int) = GET("$baseUrl/recently-added/?page=$page")
+
+    override fun latestUpdatesSelector() = popularAnimeSelector()
+
+    override fun latestUpdatesFromElement(element: Element) = popularAnimeFromElement(element)
+
+    override fun latestUpdatesNextPageSelector() = popularAnimeNextPageSelector()
+
+    override fun latestUpdatesParse(response: Response): AnimesPage {
+        aniSamaFilters.fetchFilters()
+        return super.latestUpdatesParse(response)
+    }
+
+    // =============================== Search ===============================
+    override fun getFilterList() = aniSamaFilters.getFilterList()
+
+    override suspend fun getSearchAnime(page: Int, query: String, filters: AnimeFilterList): AnimesPage {
+        return if (query.startsWith(PREFIX_SEARCH)) { // URL intent handler
+            val id = query.removePrefix(PREFIX_SEARCH)
+            client.newCall(GET("$baseUrl/anime/$id")).awaitSuccess().use(::searchAnimeByIdParse)
+        } else {
+            super.getSearchAnime(page, query, filters)
+        }
+    }
+
+    private fun searchAnimeByIdParse(response: Response): AnimesPage {
+        val details = animeDetailsParse(response.asJsoup())
+            .apply { setUrlWithoutDomain(response.request.url.toString()) }
+        return AnimesPage(listOf(details), false)
+    }
+
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList): Request {
+        val url = "$baseUrl/filter".toHttpUrl().newBuilder()
+        url.addQueryParameter("keyword", query)
+        url.addQueryParameter("page", page.toString())
+        filters.filterIsInstance<AniSamaFilters.QueryParameterFilter>().forEach {
+            val (name, value) = it.toQueryParameter()
+            if (value != null) url.addQueryParameter(name, value)
+        }
+        return GET(url.build())
+    }
+
+    override fun searchAnimeSelector() = popularAnimeSelector()
+
+    override fun searchAnimeFromElement(element: Element) = popularAnimeFromElement(element)
+
+    override fun searchAnimeNextPageSelector() = popularAnimeNextPageSelector()
+
+    // =========================== Anime Details ============================
+    private fun Elements.getMeta(name: String) = select(".item:has(.item-title:contains($name)) > .item-content")
+
+    private fun parseStatus(status: String) = when (status) {
```

Comment:
Make this a function under `Element` like how you did with `.getMeta()` for consistency's sake